test/conftest.py

```diff
@@ -0,0 +1,89 @@
+#  Copyright (c) University College London Hospitals NHS Foundation Trust
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""System/E2E test setup"""
+import logging
+import subprocess
+from pathlib import Path
+
+import pytest
+
+pytest_plugins = "pytest_pixl"
+
+
+@pytest.fixture()
+def host_export_root_dir():
+    """Intermediate export dir as seen from the host"""
+    return Path(__file__).parents[1] / "exports"
+
+
+def run_subprocess(
+    cmd: list[str], working_dir: Path, *, shell=False, timeout=360
+) -> subprocess.CompletedProcess[bytes]:
+    """
+    Run a command but capture the stderr and stdout better than the CalledProcessError
+    string representation does
+    """
+    try:
+        logging.info("Running command %s", cmd)
+        return subprocess.run(
+            cmd,
+            check=True,
+            cwd=working_dir,
+            shell=shell,  # noqa: S603 input is trusted
+            timeout=timeout,
+            capture_output=True,
+        )
+    except subprocess.CalledProcessError as exception:
+        logging.error("*** exception occurred running: '%s'", cmd)  # noqa: TRY400 will raise anyway
+        logging.error("*** stdout:\n%s", exception.stdout.decode())  # noqa: TRY400
+        logging.error("*** stderr:\n%s", exception.stderr.decode())  # noqa: TRY400
```

Comment:
Have you looked at whether logging.exception() would do the job here?

Reply:
The exception is being re-raised so will get printed eventually. The problem is that the stderr and stdout strings get shortened - only the beginning and end is shown - hence manually logging in their entirety here. I don't think I tried it but I'd expect `logging.exception` to have the same problem.